README.md

```diff
@@ -1,22 +1,119 @@
-# TAP::Harness
+# NAME
 
-This is an asynchronous TAP framework writen in Perl 6
+TAP::Harness
 
-# Description
+# DESCRIPTION
 
-This module runs TAP test files and parses them asynchronously.
+An asynchronous TAP framework written in Perl 6.
 
-# How to use
+# prove6
 
-bin/prove6 is the easiest way to use it. It's fairly basic but similar enough to p5's prove, e.g.
+This module provides the `prove6` command which runs a TAP based test suite and prints a report.
+The `prove6` command is a minimal wrapper around an instance of this module.
 
- prove6 -l t/basic.t
+For a detailed documentation on how to use `prove6` [click here](./README-prove6.md) or run `prove6 --help`
+
+# SYNOPSIS
+
+```Perl 6
+use TAP::Harness;
+my $harness = TAP::Harness.new(|%args);
+$harness.run(@tests);
+```
+
+# METHODS
+
+## Class Methods
+
+### new
+
+```Perl 6
+my %args = jobs => 1, err  => 'ignore';
+my $harness = TAP::Harness.new( |%args );
+```
+
+The constructor returns a new `TAP::Harness` object.
+It accepts an optional hash whose allowed keys are:
+
+* `volume`
+
+  Default value: `Normal`
+
+  Possible values: `Silent` `ReallyQuiet` `Quiet` `Normal` `Verbose`
+* `jobs`
+
+  The maximum number of parallel tests to run.
+
+  Default value: `1`
+
+  Possible values: An `Int`
+* `timer`
+
+  Append run time for each test to output.
+
+  Default value: `False`
+
+  Possible values: `True` `False`
+* `err`
+
+  Error reporting configuration.
+
+  Default value: `stderr`
+
+  Possible values: `stderr` `ignore` `merge`
```

Comment:
Actually, there are two more possibilities: it can also be either a Supply or an IO::Handle.

Reply:
Should be taken care by 89096ad